tool/runner/TypeDBCoreRunner.java

```diff
@@ -32,17 +38,24 @@ public class TypeDBCoreRunner implements TypeDBRunner {
     private final Path dataDir;
     private final Path logsDir;
     private final int port;
+    private final int diagnosticsMonitoringPort;
     private StartedProcess process;
     private final ProcessExecutor executor;
-    private Map<String, String> options;
+    private final Map<String, String> userOptions;
+
+    private static final Map<String, String> OVERRIDABLE_OPTIONS = map(
+            pair(DEVELOPMENT_MODE_ENABLE, "true")
+    );
 
     public TypeDBCoreRunner() throws InterruptedException, TimeoutException, IOException {
         this(new HashMap<>());
     }
 
-    public TypeDBCoreRunner(Map<String, String> options) throws InterruptedException, TimeoutException, IOException {
-        this.options = options;
-        port = findUnusedPorts(1).get(0);
+    public TypeDBCoreRunner(Map<String, String> userOptions) throws InterruptedException, TimeoutException, IOException {
+        this.userOptions = userOptions;
+        List<Integer> unusedPorts = findUnusedPorts(2);
+        port = unusedPorts.get(0);
+        diagnosticsMonitoringPort = unusedPorts.get(1);
```

Comment:
I add `diagnostics.monitoring.port` generation for parallel `TypeDBCoreRunner` runs, otherwise, they would conflict with each other just like they did in `Cloud` if we didn't turn `diagnostics.monitoring` off.